pandas/tests/groupby/test_reductions.py

```diff
@@ -1057,3 +1057,27 @@ def test_regression_allowlist_methods(op, axis, skipna, sort):
         if sort:
             expected = expected.sort_index(axis=axis)
         tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("multiplier", range(90))
+def test_prod_groupby_consistency_values(multiplier):
+    data = [
+        [1, 10],
+        [1, 21],
+        [1, 32],
+        [1, 43],
+        [1, 54],
+        [1, 67],
+        [1, 78],
+        [1, 89],
+        [1, 5],
+        [1, 17],
+        [1, 37],
+        [1, 47],
+        [1, 99],
+        [1, multiplier],
+    ]
+    df = DataFrame(data, columns=["A", "B"])
+    df_prod = df.prod()["B"]
+    df_groupby_prod = df.groupby(["A"]).prod().iloc[0, 0]
+    assert df_prod == df_groupby_prod
```

Comment:
Can you compare the whole result with assert_frame_equal?

Reply:
Hi @phofl , thanks for your reply and suggestion. I have updated the test. Hope your review.